workers/worker_git_integration.py

```diff
@@ -1263,131 +1008,114 @@ def update_gitlab_rate_limit(self, response, bad_credentials=False, temporarily_
             # Change headers to be using the new oauth's key
             self.headers = {"PRIVATE-TOKEN" : self.oauths[0]['access_token']}
 
-    def update_rate_limit(
-        self, response, bad_credentials=False, temporarily_disable=False, platform="gitlab"
-    ):
-        if platform == 'gitlab':
-            return self.update_gitlab_rate_limit(
-                response, bad_credentials=bad_credentials, temporarily_disable=temporarily_disable
+    def update_gh_rate_limit(self, response, bad_credentials=False, temporarily_disable=False):
+        # Try to get rate limit from request headers, sometimes it does not work (GH's issue)
+        #   In that case we just decrement from last recieved header count
+        if bad_credentials and len(self.oauths) > 1:
+            self.logger.warning(
+                f"Removing oauth with bad credentials from consideration: {self.oauths[0]}"
             )
-        elif platform == 'github':
```

Comment:
@IsaacMilarky :This might not work the same way as the current version. Its something I'll look for in the logs during testing.